src/useSubtitles.js

```diff
@@ -0,0 +1,114 @@
+import { getVoiceFromCue, isJSON, toIterable } from './machinery/utilities';
+import { initial, initialCurrentSubtitle } from './defaults';
+import { useCallbackRef } from "./machinery/useCallbackRef";
+import { useEvent } from "./machinery/useEvent";
+
+export function useSubtitles({ language = "nl" }) {
+  const [subtitles, setSubtitles] = React.useState({ [language]: [] });
+  const [metadata, setMetadata] = React.useState({ [language]: [] });
+  const [currentSubtitle, setCurrentSubtitle] = React.useState({ [language]: initialCurrentSubtitle });
+  const [currentMetadata, setcurrentMetadata] = React.useState({ [language]: initial });
+
+  const onMetadataChangeEvent = useEvent(handleMetadataChange);
+  const onCueChangeEvent = useEvent(handleCueChange);
+  const onMountEvent = useEvent(handleInitialLoad);
+  const onUnmountEvent = useEvent(handleUnmount);
+
+  const memoizedMetadata = React.useMemo(() => metadata, [metadata]);
+  const memoizedSubtitles = React.useMemo(() => subtitles[language], [subtitles, language]);
+  const memoizedCurrentMetadata = React.useMemo(() => currentMetadata[language], [currentMetadata, language]);
+  const memoizedCurrentSubtitle = React.useMemo(() => currentSubtitle[language], [currentSubtitle, language])
+
+  const ref = useCallbackRef({
+    onMount: onMountEvent,
+    onUnmount: onUnmountEvent
+  })
+
+  /** @param {HTMLMediaElement} node */
+  function handleInitialLoad(node) {
+    const tracks = toIterable(node?.textTracks)
+
+    // Set initial subtitles
+    node?.addEventListener('loadedmetadata', () => {
```

Comment:
Het lijkt erop dat er een soort dubbeling zit tussen subtitles en metadata. Kun je die niet gelijktrekken? Misschien is het goed om daar nog even over te sparren samen.

Reply:
Yes, in de refactor meegenomen!